### PR TITLE
VPN-3682 - add quantity metric to glean

### PR DESCRIPTION
**Pull request by mcleinman**
## Description

Adding quantity metric to glean. From some local tests, it seems to work.

## Reference

https://mozilla-hub.atlassian.net/browse/VPN-3682

## Checklist
    
- [X] My code follows the style guidelines for this project
- [X] I have not added any packages that contain high risk or unknown licenses (GPL,  LGPL, MPL, etc. consult with DevOps if in question)
- [X] I have performed a self review of my own code
- [X] I have commented my code PARTICULARLY in hard to understand areas
- [X] I have added thorough tests where needed

### CODE REVIEW

File: qtglean/include/glean/quantity.h [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  // Test  only functions
+
+  Q_INVOKABLE int64_t testGetValue(const QString& pingName = "") const;
+  Q_INVOKABLE int32_t testGetNumRecordedErrors(ErrorType errorType) const;
+
+ private:
+  int m_id;
+};
+
+#endif  // QUANTITY_H
```
Comment (brizental) [inline qtglean/include/glean/quantity.h:35]: nit: Newline.

File: qtglean/include/glean/quantity.h [hunk 2]
```diff
@@ -0,0 +1,35 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef QUANTITY_H
+#define QUANTITY_H
+#include <QObject>
```
Comment (brizental) [inline qtglean/include/glean/quantity.h:7]: nit: Newline above.

File: qtglean/src/ffi/quantity.rs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+    id: u32,
+    error_type: ErrorType,
+) -> i32 {
+    with_metric!(
+        QUANTITY_MAP,
+        id,
+        metric,
+        metric.test_get_num_recorded_errors(error_type)
+    )
+}
```
Comment (brizental) [inline qtglean/src/ffi/quantity.rs:39]: nit: Newline. This can be automated if you use VSCode, btw.

**Review by brizental — APPROVED** [review]
r+wc.

File: qtglean/include/glean/quantity.h (hunk 2 above)
Comment (mcleinman) [reply qtglean/include/glean/quantity.h:8]: Done

File: qtglean/src/ffi/quantity.rs (hunk 3 above)
Comment (mcleinman) [reply qtglean/src/ffi/quantity.rs]: Done, and updated VSCode settings

File: qtglean/include/glean/quantity.h (hunk 1 above)
Comment (mcleinman) [reply qtglean/include/glean/quantity.h]: Done

**Pull request merged**